src/Form.elm

```diff
@@ -197,14 +200,57 @@ update validation msg (F model) =
                     else
                         model.dirtyFields
 
-                newChangedFields =
-                    Set.insert name model.changedFields
+                ( newChangedFields, newOriginalValues ) =
+                    if Set.member name model.changedFields then
+                        let
+                            storedValue =
+                                Dict.get
+                                    name
+                                    model.originalValues
+                                    |> Maybe.withDefault Nothing
```

Comment:
`sameAsOriginal` should be `Bool` so there is possible use it and after finally call `Maybe.withDefault False` (it's also missing in my `Maybe.map` example above) but in this case I need returning `Just True` or `Just False` inside. 

I don't know if it's clear but let me describe what is happening here:
- Input is empty at the beginning and stored original value is `Nothing` (based on what this part returns `originalValue = (getFieldAt name model |> Maybe.andThen Tree.asValue)` (after`Input` is called)
- user provide another change on input (delete what was written before) -> current fieldValue is `Just ""` not `Nothing` so I need compare here stored `Nothing` vs `Just ""` as same values.  So I need logic in `Nothing` part of case condition for storedValue (storedValue is similar to defaultValue). Is it clear?